app/models/profile.rb

```diff
@@ -3,12 +3,47 @@ class Profile < ApplicationRecord
 
   validates :user_id, uniqueness: true
 
-  # This method generates typed accessors for all profile fields
-  def self.refresh_store_accessors!
+  USER_MIXIN = Module.new
+
+  # NOTE: @citizen428 This is a temporary mapping so we don't break DEV during
+  # profile migration/generalization work.
+  MAPPED_ATTRIBUTES = {
+    brand_color1: :bg_color_hex,
+    brand_color2: :text_color_hex,
+    display_email_on_profile: :email_public,
+    display_looking_for_work_on_profile: :looking_for_work_publicly,
+    git_lab_url: :gitlab_url,
+    linked_in_url: :linkedin_url,
+    recruiters_can_contact_me_about_job_opportunities: :contact_consent,
+    stack_overflow_url: :stackoverflow_url
+  }.with_indifferent_access.freeze
+
+  # Generates typed accessors for all currently defined profile fields.
+  def self.refresh_attributes!
     ProfileField.find_each do |field|
-      store_attribute :data, field.attribute_name, field.type
+      attribute_name = field.attribute_name
+      store_attribute :data, attribute_name, field.type
+
+      getter = MAPPED_ATTRIBUTES.fetch(attribute_name, attribute_name).to_s
```

Comment:
Here we add to the dynamic module defined above. This works because Ruby is extremly late binding and method lookup doesn't happen until runtime.

Simplied example:

```ruby
# Define a test module and class
module TestMixin; end
class Test; include TestMixin; end

# There is no hello method yet
test1 = Test.new
test1.hello
# NoMethodError: undefined method `hello' for #<Test:0x0000000110b92100>

# Let's add one
TestMixin.instance_eval { define_method(:hello) { "👋" } }

# New test instances now have a hello method
Test.new.hello
#=> "👋"

# BUT: test1 also now has a hello method, since method lookup has changed
test1.hello
#=> "👋"
```

Reply:
Well, this was a nice idea and worked, but we got rid of the proxying for now, so 👋